components/camel-jooq/src/main/java/org/apache/camel/component/jooq/JooqEndpoint.java

```diff
@@ -103,7 +104,10 @@ protected Expression createProducerExpression() {
             type = null;
         }
 
-        if (type == null) {
+        // Do not validate plain SQL
+        boolean isSQLQuery = !ObjectHelper.isEmpty(configuration.getQuery());
+
+        if (type == null || isSQLQuery) {
```

Comment:
for null checking you can use the ObjectHelper class

Reply:
Thanks, removed that condition, because validation will never be called for SQL queries.